python/shipDigiReco.py

```diff
@@ -931,34 +931,44 @@ def findTracks(self):
 #check
     atrack = entry[1]
     theTrack = entry[0]
-    if not theTrack.checkConsistency():
-     print('Problem with track before fit, not consistent',atrack,theTrack)
-     continue
+    try:
+      theTrack.checkConsistency()
+    except ROOT.genfit.Exception as e:
+      print('Problem with track before fit, not consistent',atrack,theTrack)
+      print(e.what())
+      ut.reportError(e)
 # do the fit
-    try:  self.fitter.processTrack(theTrack) # processTrackWithRep(theTrack,rep,True)
+    try:
+      self.fitter.processTrack(theTrack) # processTrackWithRep(theTrack,rep,True)
     except:
       if global_variables.debug:
         print("genfit failed to fit track")
       error = "genfit failed to fit track"
       ut.reportError(error)
       continue
 #check
-    if not theTrack.checkConsistency():
-     if global_variables.debug:
-       print('Problem with track after fit, not consistent', atrack, theTrack)
-     error = "Problem with track after fit, not consistent"
-     ut.reportError(error)
-     continue
+    try:
+      theTrack.checkConsistency()
+    except:
```

Comment:
We could narrow the exception caught here as well

Reply:
Done.